src/VisualStudio/IntegrationTest/TestUtilities/VisualStudioInstanceFactory.cs

```diff
@@ -272,12 +278,94 @@ private static Process StartNewVisualStudioProcess(string installationPath)
             IntegrationHelper.KillProcess("dexplore");
 
             var process = Process.Start(vsExeFile, VsLaunchArgs);
-
             Debug.WriteLine($"Launched a new instance of Visual Studio. (ID: {process.Id})");
 
             return process;
         }
 
+        // Starts ProcDump against the processId.
+        // Returns the path to the dump directory.
+        private static string StartProcDump(int processId)
+        {
+            Debug.WriteLine("Ensuring ProcDump");
+            var procDumpFilePath = EnsureProcDump();
+            Debug.WriteLine("Ensured ProcDump");
+            var dumpDirectory = CreateDumpDirectory();
+            var currentDumpDirectory = Path.Combine(dumpDirectory, Guid.NewGuid().ToString());
+            if (Directory.EnumerateFiles(dumpDirectory, "*.*", SearchOption.AllDirectories).Count() < MaxDumpCount)
+            {
+                if (!Directory.Exists(currentDumpDirectory))
+                {
+                    Directory.CreateDirectory(currentDumpDirectory);
+                }
+
+                var procDumpProcess = Process.Start(procDumpFilePath, $" /accepteula -ma -e -h -t -w {processId} {currentDumpDirectory}");
+                Debug.WriteLine($"Launched ProcDump attached to {processId}");
+            }
+            else
+            {
+                Debug.WriteLine($"Dump directory created but number of dumps created by previous tests exceed the max dump count limit. No more dumps will be created during the session.");
+            }
+
+            return currentDumpDirectory;
+        }
+
+        // Ensure that procdump is available on the machine. 
+        // Returns the path to the file that contains the procdump binaries (both 32 and 64 bit)
+        private static string EnsureProcDump()
```

Comment:
yes, removed as agreed. we will assume that the script was executed. otherwise, we will not start procdump

---
In reply to: [121031140](https://github.com/dotnet/roslyn/pull/20129#discussion_r121031140) [](ancestors = 121031140)